src/bson-encode.c

```diff
@@ -410,7 +410,7 @@ static void php_phongo_bson_append(bson_t *bson, php_phongo_bson_flags_t flags,
 #endif
 
 		default:
-			phongo_throw_exception(PHONGO_ERROR_UNEXPECTED_VALUE TSRMLS_CC, "Got unsupported type %d '%s'", Z_TYPE_P(entry), zend_get_type_by_const(Z_TYPE_P(entry)));
+			phongo_throw_exception(PHONGO_ERROR_UNEXPECTED_VALUE TSRMLS_CC, "Detected unsupported PHP type %d (%s) for fieldname \"%s\"", Z_TYPE_P(entry), zend_get_type_by_const(Z_TYPE_P(entry)), key);
```

Comment:
ditto - and also the value is not in " ... " here.

Reply:
There isn't a value to report here, as the zval type is unknown.